src/_png.cpp

```diff
@@ -408,13 +408,21 @@ static void _read_png_data(PyObject *py_file_obj, png_bytep data, png_size_t len
     Py_ssize_t bufflen;
     if (read_method) {
         result = PyObject_CallFunction(read_method, (char *)"i", length);
-        if (PyBytes_AsStringAndSize(result, &buffer, &bufflen) == 0) {
-            if (bufflen == (Py_ssize_t)length) {
-                memcpy(data, buffer, length);
-            } else {
-                PyErr_SetString(PyExc_IOError, "read past end of file");
-            }
-        }
+	if (result) {
```

Comment:
All the code uses some mixed tab/space indent. It should be 4-space indent even in the C code, I believe.